libraries/AP_RangeFinder/AP_RangeFinder_MAVLink.cpp

```diff
@@ -45,28 +45,31 @@ void AP_RangeFinder_MAVLink::handle_msg(const mavlink_message_t &msg)
     }
 }
 
-int16_t AP_RangeFinder_MAVLink::max_distance_cm() const
+float AP_RangeFinder_MAVLink::max_distance() const
 {
-    if (_max_distance_cm == 0 && _min_distance_cm == 0) {
+    const auto baseclass_max_distance = AP_RangeFinder_Backend::max_distance();
+
+    if (is_zero(_max_distance) && is_zero(_min_distance)) {
         // we assume if both of these are zero that we ignore both
-        return params.max_distance_cm;
+        return baseclass_max_distance;
     }
 
-    if (params.max_distance_cm < _max_distance_cm) {
-        return params.max_distance_cm;
-    }
-    return _max_distance_cm;
+    // return the larger of the base class's distance and what we
+    // receive from the network:
+    return MAX(baseclass_max_distance, _max_distance);
 }
-int16_t AP_RangeFinder_MAVLink::min_distance_cm() const
+float AP_RangeFinder_MAVLink::min_distance() const
 {
-    if (_max_distance_cm == 0 && _min_distance_cm == 0) {
+    const auto baseclass_min_distance = AP_RangeFinder_Backend::min_distance();
+
+    if (is_zero(_max_distance) && is_zero(_min_distance)) {
         // we assume if both of these are zero that we ignore both
-        return params.min_distance_cm;
+        return baseclass_min_distance;
     }
-    if (params.min_distance_cm > _min_distance_cm) {
-        return params.min_distance_cm;
-    }
-    return _min_distance_cm;
+
+    // return the smaller of the base class's distance and what we
+    // receive from the network:
+    return MIN(baseclass_min_distance, _min_distance);
```

Comment:
I would have expected this one to be `MAX`

Reply:
Fixed.  I think I recall us discussing this change at DevCall, but I can't find nnotes on it, so I've made it work like the old code.